pinc/metarefresh.inc

```diff
@@ -59,12 +59,13 @@ function metarefresh($seconds, $url, $title="", $body="", $allow_external=FALSE)
     // If we can't do the redirection using HTTP headers (which is the common
     // case), fall back to using a <meta> tag.
 
-    // See if ambersands are present and if not escaped, escape them
-    if(strpos($url, '&') !== FALSE && strpos($url, '&amp;') === FALSE) {
-        $url = str_replace('&', '&amp;', $url);
+    // If ampersands are present and already escaped, unescape them so
+    // attr_safe() will re-escape them along with anything else
+    if(strpos($absolute_url, '&amp;') !== FALSE) {
+        $absolute_url = str_replace('&amp;', '&', $absolute_url);
```

Comment:
Should we find the callers to this method and try to remove the encoding?

Reply:
We could, but I'm not sure it's worth the effort.